### PR TITLE
Reminder

### PR DESCRIPTION
**Pull request by gopalp1**
reminder table layout asset  layout

### CODE REVIEW

File: src/app/asset-management/components/asset-manage/add/add.component.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+   if(data.returnedProduct === true){
+
+   const config = {
+    url:urls.assets_assign.return  + this.assigned_asset?._id  + "/" + data.product ,
+    // payload:{
+    //   returnedProduct:data.returnedProduct
+    // }
+   }
+   this.apiService.put(config).subscribe((resp)=>{
+    console.log(resp)
```
Comment (ecampusStreet) [inline src/app/asset-management/components/asset-manage/add/add.component.ts:126]: Console.log remove

File: src/app/asset-management/components/asset-manage/add/add.component.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      }
+    })
+  }
+
+  getAsset(){
+    const config = {
+      url: urls.assets_assign.getById + this.id,
+    }
+    this.apiService.get(config).subscribe((resp) => {
+      console.log(resp,"response 0")
```
Comment (ecampusStreet) [inline src/app/asset-management/components/asset-manage/add/add.component.ts:156]: Remove console logs

File: src/app/asset-management/components/asset-manage/list/list.component.html [hunk 3]
```diff
@@ -0,0 +1,78 @@
+<div class="flex-auto">
+    <div class="flex flex-col">
+
+        <div class="min-h-screen  assign-asset ">
+            <div class="flex justify-between items-center mx-8 mt-4">
+                <div class="flex items-center">
+                    <button class="btn-button " mat-raised-button (click)="add()">
+                        Assign Asset
```
Comment (ecampusStreet) [inline src/app/asset-management/components/asset-manage/list/list.component.html:8]: Assign asset

File: src/app/asset-management/components/asset-manage/list/list.component.html [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                        </tr> 
+                    </tbody>
+                    </table>
+                </div>
+            </div>
+
+
+            </div>
+
+            <!-- <div class="flex justify-center ml-3 mt-3 mb-3 mr-3">
```
Comment (ecampusStreet) [inline src/app/asset-management/components/asset-manage/list/list.component.html:66]: Remove unwanted code

File: src/app/asset-management/components/asset-manage/list/list.component.scss [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+   .btn-button {
+      background-color: #0ea5e9;
+      border-radius: 27px;
+      padding: 1px 30px;
+      color: white;
+      font-weight:500 ;
+   }
+
+   table {
```
Comment (ecampusStreet) [inline src/app/asset-management/components/asset-manage/list/list.component.scss:12]: use this as common css

File: src/app/asset-management/components/asset-manage/list/list.component.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    this.page = 1;
+    this.assetAssignedList= [];
+    this.getAssets();
+  }
+  getAssets(){
+    const config ={
+      url:urls.assets_assign.list + '?limit=' + this.limit + '&page=' + this.page + '&search=' + this.searchText ,
+    }
+    this.apiService.get(config).subscribe((resp)=>{
+      console.log(resp,'response143')
```
Comment (ecampusStreet) [inline src/app/asset-management/components/asset-manage/list/list.component.ts:40]: logs

File: src/app/asset-management/components/asset-manage/list/list.component.ts [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+      if(resp.success){
+        this.assetAssignedList =resp.result.data;
```
Comment (ecampusStreet) [inline src/app/asset-management/components/asset-manage/list/list.component.ts:42]: you to push the objects into array

File: src/app/asset-management/components/assets/add/add.component.html [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+                    <div class="back-btn" >
+                        <mat-icon (click)="back()"> arrow_back </mat-icon>
+                      </div>
+                  <div class="p-2 clear-both mr-8 ml-8 mt-16">
+                  <form [formGroup]="asset" *ngIf="showForm">
+                      <div class="flex ">
+                        <div class="w-full">
+                            <div class="grid grid-rows-2 grid-flow-col gap-2">
+                                    <mat-form-field class="example-full-width " appearance="outline">
+                                        <mat-label >Asset Name</mat-label>
```
Comment (ecampusStreet) [inline src/app/asset-management/components/assets/add/add.component.html:17]: Asset name

File: src/app/asset-management/components/assets/add/add.component.html [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                                          formControlName="stockIn" numbersOnly  type="number">
+                                      </mat-form-field>
+                                      <mat-form-field appearance="outline">
+                                        <mat-label>Status</mat-label>
+                                        <mat-select  matNativeControl name="status"  formControlName="status">
+                                          <mat-option *ngFor="let status of statuses" [value]="status?.value" >
+                                            {{status?.label}}
+                                          </mat-option>
+                                        </mat-select>
+                                        <mat-error>Please Select branch status.</mat-error>
```
Comment (ecampusStreet) [inline src/app/asset-management/components/assets/add/add.component.html:34]: Please select branch status.

File: src/app/asset-management/components/assets/add/add.component.ts [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+  back(){
+    this.location.back();
+  }
+
+  getAsset(){
+    const config ={
+      url:urls.assets.getById + this.id,
+    }
+    this.apiService.get(config).subscribe((resp)=>{
+      console.log(resp,"resp")
```
Comment (ecampusStreet) [inline src/app/asset-management/components/assets/add/add.component.ts:72]: logs

File: src/app/asset-management/components/assets/list/list.component.html [hunk 11]
```diff
@@ -0,0 +1,86 @@
+<div class="flex-auto">
+    <div class="flex flex-col">
+
+        <div class="min-h-screen  asset-box ">
+            <div class="flex justify-between items-center mx-8 mt-4">
+                <div class="flex items-center">
+                    <button class="btn-button " mat-raised-button (click)="add()">
+                        Add Asset
```
Comment (ecampusStreet) [inline src/app/asset-management/components/assets/list/list.component.html:8]: Add asset

File: src/app/asset-management/components/assets/list/list.component.html [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+                        <thead>
+                            <tr>
+                                <th class ="left-row" >ASSET NAME</th>
+                                <th class ="center-row" >DESCRIPTION</th>
+                                <th class="center-row">PRODUCT</th>
+                                <th class ="right-row" >ACTION</th>
+                                
+                            </tr>
+                        </thead>
+                    <tbody class = "gfg">
```
Comment (ecampusStreet) [inline src/app/asset-management/components/assets/list/list.component.html:35]: give proper class names

File: src/app/asset-management/components/assets/list/list.component.html [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+                            </td>     
+                        </tr> 
+                    </tbody>
+                    </table>
+                </div>
+
+
+            </div>
+
+            <!-- <div class="flex justify-center ml-3 mt-3 mb-3 mr-3">
```
Comment (ecampusStreet) [inline src/app/asset-management/components/assets/list/list.component.html:66]: remove unwanted code

File: src/app/asset-management/components/assets/list/list.component.scss [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+   .btn-button {
+      background-color: #0ea5e9;
+      border-radius: 27px;
+      padding: 1px 30px;
+      color: white;
+      font-weight:500 ;
+   }
+
+   table {
```
Comment (ecampusStreet) [inline src/app/asset-management/components/assets/list/list.component.scss:12]: use Table css as common css

File: src/app/asset-management/components/assets/list/list.component.ts [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+
+  }
+  getAssets(){
+    const config = {
+      url:urls.assets.list,
+    }
+    this.apiService.get(config).subscribe((resp)=>{
+      if(resp.success){
+        this.assetList= resp.result.data;
+      console.log(this.assetList,"this.assetList")
```
Comment (ecampusStreet) [inline src/app/asset-management/components/assets/list/list.component.ts:38]: logs

File: src/app/asset-management/components/assets/list/list.component.ts [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+  ngOnInit(): void {
+    this.getAssets();
+  }
+
+  search(){
+
+  }
+  getAssets(){
+    const config = {
+      url:urls.assets.list,
```
Comment (ecampusStreet) [inline src/app/asset-management/components/assets/list/list.component.ts:33]: pagination?

File: src/app/asset-management/components/assets/list/list.component.ts [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+      case 'view':
+        this.router.navigate(['expertizzy/asset-management/assetAdd'],{queryParams:{id :data._id, readOnly:true}});
+        break;
+    }
+  }
+
+  deleteConfirmationpopup(event: any) {
+    let data = {
+      header: '',
+      message: 'are you sure, you want to delete this asset?',
```
Comment (ecampusStreet) [inline src/app/asset-management/components/assets/list/list.component.ts:69]: Are not are

File: src/app/helpdesk/add/add.component.html [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+                                          </mat-form-field>
+
+
+                                        <mat-form-field class="example-full-width row-span-2 ml-4" appearance="outline">
+                                            <mat-label>Description</mat-label>
+                                            <textarea matInput placeholder="Please enter description" cols="40"
+                                                name="description" formControlName="description"
+                                                style="height:114px"></textarea>
+                                                <mat-error *ngIf="f['description'].invalid &&( f['description'].touched || f['description'].dirty  )" class="text-red-500">
+                                                    <mat-error *ngIf="f['description'].errors && f['description'].errors['required']">description  is required.
```
Comment (ecampusStreet) [inline src/app/helpdesk/add/add.component.html:45]: Description  is required.

File: src/app/helpdesk/add/add.component.ts [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+      }
+    })
+  }
+  submit(){
+  const config ={
+    url:urls.helpdesk.create,
+    payload:this.helpDesk.value
+  }
+  this.apiService.post(config).subscribe((resp)=>{
+    console.log(resp,"response of submit")
```
Comment (ecampusStreet) [inline src/app/helpdesk/add/add.component.ts:97]: logs

File: src/app/helpdesk/add/add.component.ts [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+
+  getdata(){
+    const config = {
+      url: urls.helpdesk.getById + this.id,
+    };
+    this.apiService.get(config).subscribe((resp) => {
+      console.log(resp,"resppp");
+      if (resp.success) {
+        this.data = resp.result.data;
+      console.log(this.data,'this.data1')
```
Comment (ecampusStreet) [inline src/app/helpdesk/add/add.component.ts:129]: logs

File: src/app/helpdesk/list/list.component.html [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+                        </tr> 
+                    </tbody>
+                    </table>
+                </div>
+
+
+
+            </div>
+
+            <!-- <div class="flex justify-center ml-3 mt-3 mb-3 mr-3">
```
Comment (ecampusStreet) [inline src/app/helpdesk/list/list.component.html:75]: remove code

File: src/app/helpdesk/list/list.component.ts [hunk 22]
```diff
@@ -0,0 +1,131 @@
...
+    const config = {
+      url:urls.helpdesk.list 
+      // + '&status=' +
+      // this.searchText + '?limit=' +
+      // this.limit +
+      // '&page=' +
+      // this.page ,
+    }
+    this.apiService.get(config).subscribe((resp)=>{
+      console.log(resp,'hjhh')
```
Comment (ecampusStreet) [inline src/app/helpdesk/list/list.component.ts:63]: logs

File: src/app/helpdesk/list/list.component.ts [hunk 23]
```diff
@@ -0,0 +1,131 @@
...
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result) {
+      }
+    });
+  }
+
+  deleteConfirmationpopup(event:any){
+    let data = {
+      header: '',
+      message: 'are you sure, you want to delete this request?',
```
Comment (ecampusStreet) [inline src/app/helpdesk/list/list.component.ts:99]: are -> Are request? -> ticket ?

File: src/app/helpdesk/requests/requests.component.html [hunk 24]
```diff
@@ -0,0 +1,85 @@
...
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <tr>
+                            <td *ngFor="let list of manageList">
+                                <mat-card-content >
+                                   <div class="flex justify-between items-center font-medium bg-white  rounded-md px-4 mt-2 table-data">
+                                    <span class="w-3/12">
+                                        {{list?.requestedBy}}
+                                        <!-- {{list.requestedBy?.personalDetails[0]?.firstName | titlecase}} {{list.requestedBy?.personalDetails[0]?.fatherName | titlecase}} {{list.requestedBy?.personalDetails[0]?.lastName | titlecase}} -->
```
Comment (ecampusStreet) [inline src/app/helpdesk/requests/requests.component.html:42]: remove

File: src/app/helpdesk/requests/requests.component.html [hunk 25]
```diff
@@ -0,0 +1,85 @@
...
+                                    </span>
+                              
+                                        <span class="w-3/12">  {{list?.subject}}</span>
+                                        <span class="w-3/12 truncate  overflow-hidden">{{list?.description}}</span>
+                                        <span class="w-2/12"> {{list?.raisedDate  | date: 'dd/MM/yyyy'}}</span>
+                                        <span class="w-1/12"> 
+                                            <!-- <button
+                                            mat-icon-button
```
Comment (ecampusStreet) [inline src/app/helpdesk/requests/requests.component.html:50]: remove

File: src/app/helpdesk/requests/requests.component.html (hunk 24 above)
Comment (ecampusStreet) [reply src/app/helpdesk/requests/requests.component.html:42]: if not needed

File: src/app/helpdesk/requests/requests.component.html (hunk 25 above)
Comment (ecampusStreet) [reply src/app/helpdesk/requests/requests.component.html:50]: if not needed

File: src/app/helpdesk/requests/requests.component.html [hunk 26]
```diff
@@ -0,0 +1,85 @@
...
+                                    </div>
+                                </mat-card-content>
+                            </td>
+                        </tr>
+                    </tbody>
+                </table>
+
+            </div>
+
+            <!-- <div class="flex justify-center ml-3 mt-3 mb-3 mr-3">
```
Comment (ecampusStreet) [inline src/app/helpdesk/requests/requests.component.html:71]: remove if not needed

File: src/app/helpdesk/requests/requests.component.ts [hunk 27]
```diff
@@ -0,0 +1,55 @@
...
+    private toastService: ToastService,
+    private userService: CurrentUserService,
+
+  ) { }
+
+  ngOnInit(): void {
+    this.getUserId();
+
+  }
+  search(){
```
Comment (ecampusStreet) [inline src/app/helpdesk/requests/requests.component.ts:25]: ??????

File: src/app/helpdesk/requests/requests.component.ts [hunk 28]
```diff
@@ -0,0 +1,55 @@
...
+    
+    });
+  }
+
+  getRequestList(){
+    const config = {
+      url:urls.helpdesk.manage + this.empID
+    }
+    this.apiService.get(config).subscribe((resp)=>{
+      console.log(resp,'response')
```
Comment (ecampusStreet) [inline src/app/helpdesk/requests/requests.component.ts:45]: logs

File: src/app/reminder/add/add.component.html [hunk 29]
```diff
@@ -0,0 +1,40 @@
+<form [formGroup]="reminder">
+    <div class="p-2 bg-white  popup">
+      <div class="flex">
+        <div class="w-full">
+          <div class="grid lg:grid-cols">
+            <mat-form-field class="example-full-width" appearance="outline">
+                <mat-label>Title</mat-label>
+                <input matInput placeholder="Please enter title " name="title" formControlName="title" />
+                <mat-error>Please provide date </mat-error>
```
Comment (ecampusStreet) [inline src/app/reminder/add/add.component.html:9]: Place holder error message not matching

File: src/app/reminder/add/add.component.ts [hunk 30]
```diff
@@ -0,0 +1,68 @@
...
+})
+export class AddComponent implements OnInit {
+  reminder!: FormGroup;
+  constructor(
+    public dialogRef: MatDialogRef<AddLeavetypeComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    private apiService: ApiService,
+    private toast: ToastService
+  ) { 
+    console.log(data,"data555")
```
Comment (ecampusStreet) [inline src/app/reminder/add/add.component.ts:20]: logs

File: src/app/reminder/list/list.component.html [hunk 31]
```diff
@@ -0,0 +1,63 @@
+<div class="flex-auto">
+    <div class="flex flex-col">
+
+        <div class="min-h-screen  reminder-box ">
+            <div class="flex justify-between items-center mx-8 mt-4">
+                <div class="">
+                    <button class="btn-button " mat-raised-button (click)="openPopUp()">
+                        Add Reminder
```
Comment (ecampusStreet) [inline src/app/reminder/list/list.component.html:8]: Add reminder

File: src/app/reminder/list/list.component.ts [hunk 32]
```diff
@@ -0,0 +1,96 @@
...
+      data: data,
+    });
+    dialogRef.afterClosed().subscribe((result) => {
+      console.log(result,'result')
+      if (result) {
+        this.getList();
+      }
+    });
+  }
+  search(){
```
Comment (ecampusStreet) [inline src/app/reminder/list/list.component.ts:39]: ???

File: src/app/reminder/list/list.component.ts [hunk 33]
```diff
@@ -0,0 +1,96 @@
...
+
+  }
+
+  getList(){
+    const config ={
+      url:urls.reminder.list,
```
Comment (ecampusStreet) [inline src/app/reminder/list/list.component.ts:45]: pagination

File: src/app/reminder/list/list.component.ts [hunk 34]
```diff
@@ -0,0 +1,96 @@
...
+      case 'delete':
+        this.deleteConfirmationpopup(data);
+        break; 
+    }
+  }
+
+  deleteConfirmationpopup(event:any){
+    let data = {
+      header: '',
+      message: 'are you sure, you want to delete this reminder?',
```
Comment (ecampusStreet) [inline src/app/reminder/list/list.component.ts:67]: Are

File: src/app/shared/components/table/table.component.html [hunk 35]
```diff
@@ -0,0 +1,58 @@
+<div class="table">
+    <div class=" mt-4">
+        <div class="flex justify-between mx-8">
+            <div class="">
+                <button class="btn-button " mat-raised-button>
+                    Add Reminder
```
Comment (ecampusStreet) [inline src/app/shared/components/table/table.component.html:6]: Add reminder

File: src/app/helpdesk/list/list.component.ts [hunk 36]
```diff
@@ -0,0 +1,131 @@
...
+    this.router.navigate(['/expertizzy/helpdesk/add']);
+  }
+  requests(){
+    this.router.navigate(['/expertizzy/helpdesk/requests']);
+  }
+
+  getList(){
+    const config = {
+      url:urls.helpdesk.list 
+      // + '&status=' +
```
[the remark this answers is not in the document]
Comment (gopalp1) [reply src/app/helpdesk/list/list.component.ts:56]: it was not working thats why i commented

File: src/app/asset-management/components/asset-manage/add/add.component.html [hunk 37]
```diff
@@ -0,0 +1,85 @@
...
+                <mat-card class=" bg-white  px-2 relative rounded-3xl " >
+                  <app-content-navbar [isBack]="true" ></app-content-navbar>
+                  <div class="p-2 clear-both mr-8 ml-8 mt-8">
+                  <form [formGroup]="asset_assign" *ngIf="showForm">
+                      <div class="flex ">
+                        <div class="w-full">
+                            <div class="grid grid-cols-1 gap-2">
+                                   
+                                      <mat-form-field appearance="outline">
+                                        <mat-label>Employee Name</mat-label>
```
Comment (ecampusStreet) [inline src/app/asset-management/components/asset-manage/add/add.component.html:16]: Employee name

**Pull request merged**